### PR TITLE
feat: Use localStorage as fall back for saving projects

### PR DESCRIPTION
**Pull request by Mitcheljager**
We've had some reports of projects seemingly not saving after coming back to the editor. This PR doesn't fix that necessarily, but it adds a fall back. The project will now also be saved to your local storage, along with a timestamp. When fetching a project your local storage is also fetched. If the timestamp of the local storage project is newer than that of the fetched project, something may have gone wrong previously. In this case we use the content of the local storage project instead.

I thought about adding some alert along the lines of "Your local save is newer than the save on the server, do you want to use this instead?", similar to steam back ups saves perhaps, but it felt like this was too convoluted and most people won't have a clue what that means. So I decided to keep it stealthy and assume that the latest save is always the desired save.

This PR also fixes a missing import, but that's unrelated.

### CODE REVIEW

**Comment by CactusPuppy** [comment]
I would agree that the latest timestamp save is probably the best option to load, but maybe something along the lines of "We recovered a version of your project that wasn't fully saved." would help indicate that something is a bit off behind the scenes?

File: app/javascript/src/utils/project.js [hunk 1]
```diff
@@ -44,12 +44,22 @@ export function createDemoProject(title) {
...
       if (!data) throw Error("No results")
 
       const parsedData = JSON.parse(data)
 
+      // If the project in localStorage is newer than the project from the API
+      // something may have gone wrong when saving to the API, but the localStorage
+      // was still saved as expected. In this case we use the data from localStorage.
+      // This is a fallback and should not be the norm.
+      if (localProject && new Date(parsedData.updated_at) < new Date(localProject.updated_at)) {
+        parsedData.content = localProject.content
```
Comment (CactusPuppy) [inline app/javascript/src/utils/project.js:60]: I wonder if we could send an event over to Bugsnag here or something so we can track how often this incident happens?

File: app/javascript/src/components/editor/Modals/CreateProjectModal.svelte [hunk 2]
```diff
@@ -1,7 +1,7 @@
 <script>
   import Modal from "./Modal.svelte"
   import { currentProject, isSignedIn, modal } from "../../../stores/editor"
-  import { createProject, renameCurrentProject } from "../../../utils/project"
+  import { createProject, renameCurrentProject, setUrl } from "../../../utils/project"
```
Comment (CactusPuppy) [inline app/javascript/src/components/editor/Modals/CreateProjectModal.svelte:4]: Pretty sure this doesn't need to be here?

Comment (Mitcheljager) [reply app/javascript/src/components/editor/Modals/CreateProjectModal.svelte:4]: It does, there was a missing import

File: app/javascript/src/utils/project.js (hunk 1 above)
Comment (Mitcheljager) [reply app/javascript/src/utils/project.js:60]: Good idea! We don't have bugsnag set up for javascript currently, but I will have a look

Comment (Mitcheljager) [reply app/javascript/src/utils/project.js:61]: Turns out it was fairly easy, their javascript package is pretty nice! I will add this to other parts of the editor as well, but I will save that for another PR

**Pull request merged**